File: array/2099_find_subsequence_of_length_k_with_the_largest_sum/2099_find_subsequence_of_length_k_with_the_largest_sum.py
class Solution:
    def maxSubsequence(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        vals = [[i, nums[i]] for i in range(n)]
        vals.sort(key = lambda x: -x[1])
        vals = sorted(vals[:k])
        res = [val for idx, val in vals]
        return res

File: array/2099_find_subsequence_of_length_k_with_the_largest_sum/2099_find_subsequence_of_length_k_with_the_largest_sum.go
package main

type node struct {
	id  int
	val int
}

func maxSubsequence(nums []int, k int) []int {
	var nodes []node
	for i := range nums {
		nodes = append(nodes, node{
			id:  i,
			val: nums[i],
		})
	}
	sort.Slice(nodes, func(i, j int) bool {
		return nodes[i].val > nodes[j].val
	})
	var r []int
	for i := 0; i < k; i++ {
		r = append(r, nodes[i].id)
	}
	sort.Ints(r)
	for i := range r {
		r[i] = nums[r[i]]
	}
	return r
}
